### ISSUE TITLE
Connection issue

### ISSUE THREAD

**Issue by Jerryshi68** [daltoniam/Starscream#850]
I test in 4.0.4 version (even use your simpletest example.
found two problems:
1. When remove server restart, the client use Starscream don't notice app any event.
2. If the client to call disconnect, server never get the notice, unless wait for ping-pong happens.

**Comment by philipyoungg** [comment]
Yeah. Tried server restart, client never tried to reconnect. I also didn't get disconnect event.
